Sprint 12

Mukhammed003 opened the pull request:
Разработано: 

1) реализована постраничная загрузка данных для ленты фотографий;
2) реализована функциональность лайков;
3) реализована возможность просмотра картинки в полном размере;
4) добавлена возможность выхода из аккаунта;
5) добавлено логирование ошибок в методы сервиса ImagesListService.

Commit:
develope(newFeature): разработка сетевого слоя для экрана с лентой фотографий

Разработан сервис (ImagesListService) для загрузки картинок из сети. Обновлён ImagesListViewController, добавлена новая функция: func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath).

Commit:
develope(newFeature): разработка теста на правильность работы fetchPhotosNextPage()

Разработан Unit тест для проверки правильности работы fetchPhotosNextPage().

Commit:
develope(newFeature): удаление теста ImagesListServiceTests

Тест ImagesListServiceTests был удалён.

Commit:
develope(newFeature): реализация отображения загруженных фото в ImagesListViewController.

Реализовано отображение картинок в ленте. Устранено дублирование картинок, с помощью добавления уникальных(id картинки) картинок в photos. Дополнительно реализовано отображение этих картинок в странице с зуммированием(SingleImageViewController).

Commit:
develope(newFeature): реализация логики лайка и отмены лайка.

Реализован сетевой трафик для лайка и отмены лайка картинки. Устранены возможные гонки при отправке запросов. Устранены глюки во время скролла вверх.

Commit:
develope(newFeature): реализация логики выхода из профиля.

Реализован выход из профиля через кнопку exitButton в ProfileViewController. При выходе очищаются все данные со всех сервисов(ProfileService, ProfileImageService, ImagesListService), очищается куки, очищается токен из хранилища.

Commit:
fix(Bug): фикс бага с OAuthTokenResponseBody.

Обновил все локально не изменённые файлы.

Commit:
fix(Bug): фикс использования форматтеров для времени.

Вывел все форматтеры в отдельнее перечисление чтобы иметь их в единственных экземплярах:

Kirchberg reviewed (approved):
🍏 Рекомендации (это можно исправить по желанию):

1. В ProfileViewController создание и настройка UI-элементов во viewDidLoad можно разбить на несколько приватных методов для улучшения читаемости (например, setupProfileImage, setupNameLabel, setupConstraints и т.д.). Это сделает viewDidLoad более компактным и понятным.

--- 

🍪 Для размышлений 

1. Сейчас сетевые запросы реализованы с помощью completion handlers. Можно изучить и попробовать применить более современные подходы, такие как Combine или async/await из Swift Concurrency. Это может сделать асинхронный код более читаемым и простым в поддержке.

---

Не забудь вмерджить пул-реквест в основную ветку.
Успехов в следующем задании!

Commit:
fix(Bug): фикс читаемости в ProfileViewController.

Написал новые приватные методы (для создания вьюшек, для добавление вьюшек в очередь, для установки констрейнтов для вьюшек, для настройки обновления UI при получении уведомлении в NotificationCenter, для обновления UI) для улучшения читаемости во viewDidLoad.
